src/tests/Form.test.tsx

```diff
@@ -0,0 +1,62 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { BrowserRouter } from "react-router-dom";
+import { store } from "../app/store";
+import { Provider } from "react-redux";
+import Form from "../components/Pages/Form/Form";
+import userEvent from "@testing-library/user-event";
+
+test("renders habit form", () => {
+  render(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Form />
+      </BrowserRouter>
+    </Provider>
+  );
+
+  const habitInput = screen.getByText(/Define your goal/i);
+  const schemaInput = screen.getByText(/Define schema/i);
+  const rewardsInput = screen.getByText(/Think about rewards/i);
+
+  expect(habitInput).toBeInTheDocument();
+  expect(schemaInput).toBeInTheDocument();
+  expect(rewardsInput).toBeInTheDocument();
+});
+
+test("displays rewards list after adding reward in the form", async () => {
+  render(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Form />
+      </BrowserRouter>
+    </Provider>
+  );
+
+  const input: HTMLInputElement = screen.getByTestId("reward-input");
+  const addButton = screen.getByTestId("add-reward");
+
+  userEvent.type(input, "new reward");
+  userEvent.click(addButton);
+
+  expect(screen.getByText(/new reward/i)).toBeInTheDocument();
```

Comment:
let's be more precise here where to search for that string.

Reply:
so here you would suggest looking for this string it the container displaying rewards?

Reply:
yes